synapse/util/caches/descriptors.py

```diff
@@ -370,13 +373,14 @@ def __init__(
             cache_context=cache_context,
         )
 
-        max_entries = int(max_entries * get_cache_factor_for(orig.__name__))
```

Comment:
Where are we doing the equivalent of this now?

Reply:
in the caches themselves

Reply:
Do you mean here https://github.com/matrix-org/synapse/pull/6391/files#diff-c322501e9b5bd8e15a2eb3c4400a0440R82 ? I'm not seeing anything to do per-cache factors? In fact `config.cache_factors` doesn't seem to be used outside the tests, though I may be missing something if its going via `set_cache_factor` callbacks?

Reply:
ergh this was in the setting-on-the-fly logic that got removed as it was made out of scope. fixed, and I added some tests.